libs/blocks/merch/merch.js

```diff
@@ -870,9 +870,19 @@ export async function buildCta(el, params) {
     return addCopyToClipboard(el, cta);
   }
 
-  // @see https://jira.corp.adobe.com/browse/MWPW-173470
+  /**
+   * TODO: This code block will be deprecated and removed in a future version.
+   * @see https://jira.corp.adobe.com/browse/MWPW-173470
+   * @see https://jira.corp.adobe.com/browse/MWPW-174411
+  */
   cta.onceSettled().then(() => {
-    if (getConfig()?.locale?.prefix === '/kr' && cta.value[0]?.offerType === OFFER_TYPE_TRIAL) cta.remove();
+    const prefix = getConfig()?.locale?.prefix;
+    if (shouldAllowKrTrial(el, prefix) && prefix === '/kr' && cta.value[0]?.offerType === OFFER_TYPE_TRIAL) {
+      cta.classList.remove('hidden-osi-trial-link');
+      return;
+    }
+
+    if (!shouldAllowKrTrial(el, prefix) && prefix === '/kr' && cta.value[0]?.offerType === OFFER_TYPE_TRIAL) cta.remove();
```

Comment:
Nit: the conditionals look a bit repetitive. How about a constant holding `prefix === '/kr' && cta.value[0]?.offerType === OFFER_TYPE_TRIAL`? Then you'd only need to check for `shouldAllowKrTrial` and either remove a class or the entire element.